src/SFA.DAS.LevyTransferMatching.Web/Orchestrators/PledgeOrchestrator.cs

```diff
@@ -220,6 +225,82 @@ public async Task<LocationSelectViewModel> GetLocationSelectViewModel(LocationSe
+                        app.NumberOfApprentices, app.MaxFunding, app.StandardDuration, app.StartDate);
+
+                    return new PledgeApplicationDownloadModel
+                    {
+                        DateApplied = app.CreatedOn,
+                        Status = app.Status,
+                        ApplicationId = app.Id,
+                        PledgeId = app.PledgeId,
+                        EmployerAccountName = app.EmployerAccountName,
+                        HasTrainingProvider = app.HasTrainingProvider,
+                        Sectors = app.Sectors ?? new List<string>(),
+                        AboutOpportunity = app.Details,
+                        BusinessWebsite = GetUrlWithPrefix(app.BusinessWebsite),
+                        FormattedEmailAddress = String.Join(";", app.EmailAddresses),
+                        FormattedSectors = String.Join(",", app.Sectors ?? new List<string>()),
+                        FirstName = app.FirstName,
+                        LastName = app.LastName,
+                        NumberOfApprentices = app.NumberOfApprentices,
+                        StartBy = app.StartDate,
+                        TypeOfJobRole = app.JobRole,
+                        EncodedPledgeId = _encodingService.Encode(app.PledgeId, EncodingType.PledgeId),
+                        EncodedApplicationId = _encodingService.Encode(app.Id, EncodingType.PledgeApplicationId),
+                        IsJobRoleMatch = app.IsJobRoleMatch,
+                        IsLevelMatch = app.IsLevelMatch,
+                        IsLocationMatch = app.IsLocationMatch,
+                        IsSectorMatch = app.IsSectorMatch,
+                        Duration = app.StandardDuration,
+                        EstimatedCostThisYear = affordability.EstimatedCostThisYear,
+                        Level = app.Level,
+                        TotalEstimatedCost = app.MaxFunding * app.NumberOfApprentices,
+                        AdditionalLocations = app.AdditionalLocations,
+                        SpecificLocation = app.SpecificLocation,
+                        Locations = app.Locations,
+                        PledgeLocations = app.PledgeLocations,
+                        FormattedLocations = GetCommaSeparatedListOfLocations(app.PledgeLocations, app.Locations, app.SpecificLocation, app.AdditionalLocations)
+                    };
+
+
+                })
+            };
+
+            var fileContents = _csvService.GenerateCsvFileFromModel(pledgeAppModel);
+
+            return fileContents;
+        }
+
+        private string GetCommaSeparatedListOfLocations(IEnumerable<GetApplyResponse.PledgeLocation> pledgeLocations, IEnumerable<GetApplicationsResponse.ApplicationLocation> applicationLocations, string specificLocation, string additionalLocations)
+        {
+            var listOfMatchingLocations = (from location in applicationLocations 
```

Comment:
Not entirely sure what's happening here, but if this sort of logic is required I think it'd be better in the apim layer.

Reply:
yeah so I asked Conor about how the specific locations and everything worked. So it checks to see if the application locations match the pledge and records the location in the enumerable, then if the specific and additional locations are present they are added to the enumerable and then it just joins them to give the CSV separated list of locations.